### PR TITLE
feat: use "Script Setup" in pages

### LINKED ISSUE #2936: Vue 3 update: Use "Script Setup" in pages
Similar to #2929, we need to refactor our Vue components in the "pages/" directory to use the new "Script Setup" syntax for Vue 3.

### PR DESCRIPTION
**Pull request by eddybrando**
## Changes

Similar to https://github.com/Qiskit/qiskit.org/pull/2935, this PR changes the Vue components in the "pages" directory to make them use Script Setup.

This isn't intended to work as-is and will need further changes, but is required to continue the migration to Vue 3.

Closes #2936

### CODE REVIEW

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you!! Could you take a look at my comments?

File: pages/advocates/index.vue [hunk 1]
```diff
@@ -18,35 +18,39 @@
   </main>
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
 import { mapGetters } from 'vuex'
-import { Component } from 'vue-property-decorator'
 import QiskitPage from '~/components/logic/QiskitPage.vue'
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply pages/advocates/index.vue:23]: Yes, this is mostly a reminder for reference about the "logic" pages, which will be refactored with https://github.com/Qiskit/qiskit.org/issues/2938.

File: pages/blockbuster.vue [hunk 2]
```diff
@@ -19,50 +19,47 @@
   </div>
 </template>
 
-<script lang="ts">
-import { Component } from 'vue-property-decorator'
+<script setup lang="ts">
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply pages/blockbuster.vue]: Done in https://github.com/Qiskit/qiskit.org/pull/2937/commits/b51b39ea93c9bf3aa7773893607032fe6c672c84.

File: pages/events/index.vue [hunk 3]
```diff
@@ -109,92 +109,93 @@ import {
...
+
+  return filter === 'regionFilters'
+    ? regionFilters.includes(filterValue)
+    : typeFilters.includes(filterValue)
+}
 
-  updateFilter (filter: string, filterValue: string, isSelected: boolean): void {
-    const payload = { filter, filterValue }
-    const { commit } = this.$store
+const updateFilter = (filter: string, filterValue: string, isSelected: boolean): void => {
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply pages/events/index.vue]: Done in https://github.com/Qiskit/qiskit.org/pull/2937/commits/940839e0f78d7a4513c1764eb574ddcf601f6237.

File: pages/events/index.vue [hunk 4]
```diff
@@ -109,92 +109,93 @@ import {
...
 
-  getCheckedFilters (filter: string) {
-    return (this as any)[filter]
-  }
+const noEvents = (this as any).filteredEvents.length === 0
 
-  updateWholeFilter (filter: string, filterValues: string[]): void {
-    const { commit } = this.$store
-    const payload = { filter, filterValues }
+const getCheckedFilters = (filter: string) => (this as any)[filter]
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply pages/events/index.vue:162]: Done in https://github.com/Qiskit/qiskit.org/pull/2937/commits/940839e0f78d7a4513c1764eb574ddcf601f6237.

**Review by y4izus — APPROVED** [review]

**Pull request merged**